### PR TITLE
fix(cdk-dasm): support CFN parameters for cdk-dasm

### PR DESCRIPTION
**Pull request by campionfellin**
Fixes: https://github.com/aws/aws-cdk/issues/5323

Previous behavior can be seen in that issue. New behavior is as such:

Input:

```
{
  "Parameters" : {
    "MyParam" : {
      "Description" : "Example Parameter",
      "Default": "Default Parameter"
    }
  },
  "Resources" : {
    "MyTopic": {
      "Type": "AWS::SNS::Topic",
      "Properties": {
          "DisplayName": "hello world"
      }
    }
  }
}
```

Output:

```
// generated by cdk-dasm at 2019-12-06T02:44:43.629Z

import { Stack, StackProps, Construct, Fn, CfnParameter } from '@aws-cdk/core';
import sns = require('@aws-cdk/aws-sns');

export class MyStack extends Stack {
    constructor(scope: Construct, id: string, props: StackProps = {}) {
        super(scope, id, props);
        const MyParam = new CfnParameter(this, 'MyParam', {
            description: "Example Parameter",
            default: "Default Parameter",
        });
        new sns.CfnTopic(this, 'MyTopic', {
            displayName: "hello world",
        });
    }
}
```

And when a `default` is not specified for the parameter, this uses:

`default: this.node.tryGetContext('MyParam'),`

to allow for Parameters to be passed in at `cdk synth` time.

✅Tests are included for both cases.

Signed-off-by: campionfellin <campionfellin@gmail.com>

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk-dasm/lib/dasm.ts [hunk 1]
```diff
@@ -70,6 +89,25 @@ export async function dasmTypeScript(template: Template, options: DisassemblerOp
   code.openBlock(`constructor(scope: Construct, id: string, props: StackProps = {})`);
   code.line(`super(scope, id, props);`);
 
+  // Parameter definitions
+
+  for (const param of parameters) {
+
+    code.indent(`const ${param.id} = new CfnParameter(this, '${param.id}', {`);
+
+    // Default is needed, so if not supplied in CFN, then we'll
```
Comment (eladb) [inline packages/cdk-dasm/lib/dasm.ts]: Why is default needed?

File: packages/cdk-dasm/test/__snapshots__/dasm.test.js.snap [hunk 2]
```diff
@@ -131,3 +131,37 @@ export class MyStack extends Stack {
...
+exports[`parameter without default 1`] = `
+"// generated by cdk-dasm 
+
+import { Stack, StackProps, Construct, Fn, CfnParameter } from '@aws-cdk/core';
+
+export class MyStack extends Stack {
+    constructor(scope: Construct, id: string, props: StackProps = {}) {
+        super(scope, id, props);
+        const MyParam = new CfnParameter(this, 'MyParam', {
+            default: this.node.tryGetContext('MyParam'),
```
Comment (eladb) [inline packages/cdk-dasm/test/__snapshots__/dasm.test.js.snap]: Not sure I understand why this is required..

File: packages/cdk-dasm/lib/dasm.ts (hunk 1 above)
Comment (campionfellin) [reply packages/cdk-dasm/lib/dasm.ts]: Hey @eladb ,

I couldn't get the stack to deploy or synth correctly without it:

```
MyStack: deploying...
MyStack: creating CloudFormation changeset...

 ❌  MyStack failed: ValidationError: Parameters: [MyParam] must have values
Parameters: [MyParam] must have values
```

Using these 2 commands:

```
cdk deploy -c MyParam=test
cdk deploy
```

And when using `cdk synth -c MyParam=test`, I don't see MyParam being set anywhere in the output yaml.

Is there another command I should be using to include parameters?

**Comment by campionfellin** [comment]
Have been learning a bit more about CDK lately. Going to close this and come back for more later!

**Pull request closed without merging**